gov2/cloudfront/CreateDistribution/CreateDistribution.go

```diff
@@ -0,0 +1,182 @@
+			HttpVersion:       cloudfrontTypes.HttpVersionHttp11,
+			DefaultRootObject: aws.String("index.html"),
+			Aliases: &cloudfrontTypes.Aliases{
+				Quantity: aws.Int32(1),
+				Items:    []string{domain},
+			},
+			ViewerCertificate: &cloudfrontTypes.ViewerCertificate{
+				ACMCertificateArn: aws.String(certificateSSLArn),
+				SSLSupportMethod:  cloudfrontTypes.SSLSupportMethodSniOnly,
+			},
+			CustomErrorResponses: &cloudfrontTypes.CustomErrorResponses{
+				Quantity: aws.Int32(1),
+				Items: []cloudfrontTypes.CustomErrorResponse{
+					{
+						ErrorCode:          aws.Int32(403),
+						ResponseCode:       aws.String("200"),
+						ErrorCachingMinTTL: aws.Int64(10),
+						ResponsePagePath:   aws.String("/index.html"),
+					},
+				},
+			},
+			Origins: &cloudfrontTypes.Origins{
+				Quantity: aws.Int32(1),
+				Items: []cloudfrontTypes.Origin{
+					{
+						DomainName: aws.String(originDomain),
+						Id:         aws.String(originDomain),
+						S3OriginConfig: &cloudfrontTypes.S3OriginConfig{
+							OriginAccessIdentity: aws.String("origin-access-identity/cloudfront/" + originAccessIdentityID),
+						},
+					},
+				},
+			},
+			CacheBehaviors: nil,
+			DefaultCacheBehavior: &cloudfrontTypes.DefaultCacheBehavior{
+				TargetOriginId:       aws.String(originDomain),
+				Compress:             aws.Bool(true),
+				ViewerProtocolPolicy: cloudfrontTypes.ViewerProtocolPolicyRedirectToHttps,
+				AllowedMethods: &cloudfrontTypes.AllowedMethods{
+					Quantity: aws.Int32(2),
+					Items: []cloudfrontTypes.Method{
+						cloudfrontTypes.MethodGet,
+						cloudfrontTypes.MethodHead,
+					},
+				},
+			},
+		},
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return cloudfrontResponse, nil
+}
+
+func (c *CFDistributionAPIImpl) createoriginAccessIdentity(domainName string) (string, error) {
+	ctx := context.Background()
+	oai, err := c.cloudfrontClient.CreateCloudFrontOriginAccessIdentity(ctx, &cloudfront.CreateCloudFrontOriginAccessIdentityInput{
+		CloudFrontOriginAccessIdentityConfig: &cloudfrontTypes.CloudFrontOriginAccessIdentityConfig{
+			CallerReference: aws.String(domainName),
+			Comment:         aws.String(domainName),
+		},
+	})
+	if err != nil {
+		return "", err
+	}
+	return *oai.CloudFrontOriginAccessIdentity.Id, nil
+}
+
+// main uses the AWS SDK for Go V2 to create an Amazon CloudFront distribution.
+// This example uses the default settings specified in your shared credentials
+// and config files.
+func main() {
+	sdkConfig, err := config.LoadDefaultConfig(context.TODO())
+
+	if err != nil {
+		fmt.Println("Couldn't load default configuration. Have you set up your AWS account?")
+		fmt.Println(err)
+		return
+	}
+
+	s3Client := s3.NewFromConfig(sdkConfig)
+	cloudfrontClient := cloudfront.NewFromConfig(sdkConfig)
+
+	cfDistribution := createCFDistribution(s3Client, cloudfrontClient)
+	bucketName := flag.String("bucket", "", "<EXAMPLE-BUCKET-NAME>")
+	if bucketName == nil || *bucketName == "" {
```

Comment:
You'll need to call flag.Parse() after setting up the flag or this will always fail.

Reply:
I changed the code to use flag.Parse().